api/subscriptions/subscriptions.go

```diff
@@ -387,15 +339,83 @@ func (s *Subscriptions) Close() {
 	s.wg.Wait()
 }
 
+func (s *Subscriptions) websocket(readerFunc func(http.ResponseWriter, *http.Request) (msgReader, error)) utils.HandlerFunc {
+	return func(w http.ResponseWriter, req *http.Request) error {
+		s.wg.Add(1)
+		defer s.wg.Done()
+
+		// Call the provided reader function
+		reader, err := readerFunc(w, req)
+		if err != nil {
+			return err
+		}
+
+		// Ensure cleanup when the connection closes
+		defer func() {
+			if closer, ok := reader.(interface{ Close() }); ok {
+				closer.Close()
+			}
+		}()
+
+		// Setup WebSocket connection
+		conn, closed, err := s.setupConn(w, req)
+		if err != nil {
+			logger.Debug("upgrade to websocket", "err", err)
+			return err
+		}
+		defer s.closeConn(conn, err)
+
+		// Stream messages
+		err = s.pipe(conn, reader, closed)
+		if err != nil {
+			logger.Debug("error in websocket pipe", "err", err)
+		}
+		return err
+	}
+}
+
+// handleGone is a handler for deprecated endpoints that returns HTTP 410 Gone.
+func handleGone(w http.ResponseWriter, _ *http.Request) error {
+	w.WriteHeader(http.StatusGone)
+	_, _ = w.Write([]byte("This endpoint is no longer supported."))
+	return nil
+}
+
 func (s *Subscriptions) Mount(root *mux.Router, pathPrefix string) {
 	sub := root.PathPrefix(pathPrefix).Subrouter()
 
 	sub.Path("/txpool").
 		Methods(http.MethodGet).
 		Name("subscriptions_pending_tx").
 		HandlerFunc(utils.WrapHandlerFunc(s.handlePendingTransactions))
-	sub.Path("/{subject:beat|beat2|block|event|transfer}").
+
+	sub.Path("/block").
+		Methods(http.MethodGet).
+		Name("subscriptions_block").
+		HandlerFunc(utils.WrapHandlerFunc(s.websocket(s.handleBlockReader)))
+
+	sub.Path("/event").
+		Methods(http.MethodGet).
+		Name("subscriptions_event").
```

Comment:
@otherview could we update these to same pattern as other APIs now? I think we didn't do it previously because it was one mount for all endpoints.

Eg.

```
Name("WS /subsciptions/beat")
```

One thing to look out for would be the metrics. I think for subs we expect the name to start with subscription

Reply:
Good idea! Fixed! Looks good now. 
<img width="804" alt="image" src="https://github.com/user-attachments/assets/b27e7ac4-daaa-49bf-b53c-193a78d6e4eb" />